python/ray/tempfile_services.py

```diff
@@ -0,0 +1,261 @@
+                            os.environ[RAY_OBJECT_STORE_SOCKET_NAME])
+    else:
+        return make_inc_temp(prefix='plasma_store_', dir=sockets_dir)
+
+
+def get_plasma_manager_socket_name():
+    sockets_dir = get_sockets_dir_path()
+    return make_inc_temp(prefix='plasma_manager_', dir=sockets_dir)
+
+
+def get_local_scheduler_socket_name(suffix=None):
+    """Get a socket name for local scheduler.
+
+    This function could be unsafe. The socket name may
+    refer to a file that did not exist at some point, but by the time
+    you get around to creating it, someone else may have beaten you to
+    the punch.
+    """
+    sockets_dir = get_sockets_dir_path()
+
+    if suffix is None:
+        raylet_socket_name = make_inc_temp(prefix='scheduler_',
+                                             dir=sockets_dir)
+    else:
+        raylet_socket_name = os.path.join(sockets_dir,
+                                          'scheduler_{}'.format(suffix))
+    return raylet_socket_name
+
+
+def get_random_ipython_notebook_path(port):
+    """Get a new random ipython notebook path"""
+
+    notebook_filepath = os.path.join(
+        os.path.dirname(os.path.abspath(__file__)), "WebUI.ipynb")
+    # We copy the notebook file so that the original doesn't get modified by
+    # the user.
+    notebook_name = make_inc_temp(suffix='.ipynb', prefix='ray_ui_',
+                                    dir=temp_root)
+    new_notebook_filepath = os.path.join(get_logs_dir_path(), notebook_name)
+    shutil.copy(notebook_filepath, new_notebook_filepath)
+    new_notebook_directory = os.path.dirname(new_notebook_filepath)
+    token = ray.utils.decode(binascii.hexlify(os.urandom(24)))
+    webui_url = ("http://localhost:{}/notebooks/{}?token={}"
+                 .format(port, notebook_name, token))
+    return new_notebook_directory, webui_url, token
+
+
+def get_random_temp_redis_config_path():
+    redis_config_name = make_inc_temp(prefix='redis_conf_', dir=temp_root)
+    return redis_config_name
+
+
+def new_log_files(name, redirect_output):
+    """Generate partially randomized filenames for log files.
+
+    Args:
+        name (str): descriptive string for this log file.
+        redirect_output (bool): True if files should be generated for logging
+            stdout and stderr and false if stdout and stderr should not be
+            redirected.
+
+    Returns:
+        If redirect_output is true, this will return a tuple of two
+            filehandles. The first is for redirecting stdout and the second is
+            for redirecting stderr. If redirect_output is false, this will
+            return a tuple of two None objects.
+    """
+    if not redirect_output:
+        return None, None
+
+    # Create a directory to be used for process log files.
+    logs_dir = get_logs_dir_path()
+    # Create another directory that will be used by some of the RL algorithms.
+    # TODO: Move it into where we need such a directory.
```

Comment:
When you add `TODO` items, please add your name, (e.g., I use `TODO(rkn)`) so that we know who to ask if we have questions about it.